include/rdma/fi_ext.h

```diff
@@ -54,11 +54,11 @@ extern "C" {
  * }
  */
 
-#define FI_PROV_SPECIFIC_EFA   (FI_PROV_SPECIFIC | 0xefa << 20)
+#define FI_PROV_SPECIFIC_EFA   (FI_PROV_SPECIFIC | 0xefaU << 20)
```

Comment:
I don't think you need to make 0xefa an unsigned integer literal.  Single FI_PROV_SPECIFIC is defined as 1U, it is an unsigned int, and (FI_PROV_SPECIFIC | 0xefa << 20) will therefore be of type unsigned int.

Reply:
If I did not use 0xefaU, I still got the same warning message. This can be reproduced with the following simple program. 
```
[ec2-user@ip-172-31-81-114 libfabric]$ cat fix-warn.c
#include<stdio.h>

#define FI_PROV_SPECIFIC       (1U << 31)
#define FI_PROV_SPECIFIC_EFA   (FI_PROV_SPECIFIC | 0xefa << 20)

/* EFA options */
enum {
       FI_OPT_EFA_RNR_RETRY = (int)(FI_PROV_SPECIFIC_EFA),
};

int main() {
  printf("value def is %d\n", FI_OPT_EFA_RNR_RETRY);
}

[ec2-user@ip-172-31-81-114 libfabric]$ gcc  -Wpedantic fix-warn.c
fix-warn.c:8:31: warning: enumerator value for ‘FI_OPT_EFA_RNR_RETRY’ is not an integer constant expression [-Wpedantic]
        FI_OPT_EFA_RNR_RETRY = (int)(FI_PROV_SPECIFIC_EFA),
                               ^
```